File: solutions/day_015/longest_substring_without_repeating_characters.dart
import 'dart:math' as math;
class Solution {
int lengthOfLongestSubstring(String s) {
    Set<String> substring = {};
    int maxLength = 0;
    int left = 0;
    int right = 0;
    while (right < s.length) {
        if (!substring.contains(s[right])) {
            substring.add(s[right]);
            maxLength = math.max(maxLength, substring.length);
            right++;
        } else {
            substring.remove(s[left]);
            left++;
        }
    }
    return maxLength;
}
}

File: day_15/longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        unique_substr = set()
        result = 0
        l = 0
        for k, v in enumerate(s):
            while v in unique_substr:
                unique_substr.remove(s[l])
                l += 1
            unique_substr.add(v)
            result = max(result, k-l+1)
        return result